### ISSUE TITLE
 pointer being freed was not allocated

### ISSUE THREAD

**Issue by kevingzhang** [elastos/Elastos.NET.Carrier.Nodejs.SDK#15]
Nodejs server has point free error. @liyangwood mentioned this issue before, I can reproduce it. Can someone fix it?

2018-04-19 21:35:53 - INFO    : Carrier: Carrier node created.
New friend added. The friend information:
{"name":"","userId":"61FgmHt1Zry9aCUdGo6VnY3RBiwdpnjAzf6kYnKeFVk2","description":"","gender":"","phone":"","email":"","region":"","presence":0,"status":1,"online":false}
Friend request from user[BKcvaN4jGPATojuPYyUbAZDnX6vEoizstUyKman9q9xF] with : Catch me if you can.
New friend added. The friend information:
{"name":"","userId":"BKcvaN4jGPATojuPYyUbAZDnX6vEoizstUyKman9q9xF","description":"","gender":"","phone":"","email":"","region":"","presence":0,"status":1,"online":false}
Friend [BKcvaN4jGPATojuPYyUbAZDnX6vEoizstUyKman9q9xF] connection changed to be online
Friend [61FgmHt1Zry9aCUdGo6VnY3RBiwdpnjAzf6kYnKeFVk2] connection changed to be online
receive message from 61FgmHt1Zry9aCUdGo6VnY3RBiwdpnjAzf6kYnKeFVk2 : Here I am
receive message from BKcvaN4jGPATojuPYyUbAZDnX6vEoizstUyKman9q9xF : where are you
receive message from 61FgmHt1Zry9aCUdGo6VnY3RBiwdpnjAzf6kYnKeFVk2 : catch me if you can
receive message from BKcvaN4jGPATojuPYyUbAZDnX6vEoizstUyKman9q9xF : 跑不掉
receive message from 61FgmHt1Zry9aCUdGo6VnY3RBiwdpnjAzf6kYnKeFVk2 : ==========
[ 'f_OiqvRsN5WxriRXAAAE', 'd9ffaca46d5990ec39501bcdf22ee7a1' ]
create carrier service with id = Tom
2018-04-19 21:39:49 - INFO    : Carrier: Carrier node created.
node(8665,0x7fffa5d7d380) malloc: *** error for object 0x102708150: pointer being freed was not allocated
*** set a breakpoint in malloc_error_break to debug
[nodemon] app crashed - waiting for file changes before starting...

**Comment by jmsaulnier** [comment]
I have this issue sometimes when I try to send a message to a friend (offline / online status issue?)

**Comment by kevingzhang** [comment]
Yes, the same as @jmsaulnier  answered. I can reproduce it both online and offline.
